astrocut/CubeFactory.py

```diff
@@ -0,0 +1,670 @@
+# Licensed under a 3-clause BSD style license - see LICENSE.rst
+
+from copy import deepcopy
+from datetime import date
+from pathlib import Path
+from time import monotonic
+from typing import List, Optional, Union
+import warnings
+
+import numpy as np
+from astropy.io import fits
+from astropy.table import Column, Table
+
+# Try to import the MADV_SEQUENTIAL constant from the mmap module
+# May fail on older Python versions or on Windows
+try:
+    from mmap import MADV_SEQUENTIAL
+    mmap_imported = True
+except ImportError:
+    mmap_imported = False
+
+from . import log
+from .exceptions import DataWarning, InvalidInputError
+from .utils.utils import _handle_verbose
+
+
+class CubeFactory():
+    """
+    Class for creating image cubes. This class is built to accept TESS SPOC FFI files,
+    but can be extended to work with other types of image files.
+    
+    Methods
```

Comment:
Maybe this is already the plan, but we might want to make this an abstract class, and make a new `SpocCubeFactory`, that, along with `TicaCubeFactory` could subclass from.